### PR TITLE
physicalplan: make OrderedAggregate work in the presence of dynamic columns

### PR DESCRIPTION
**Pull request by asubiotto**
This required adding some more fields to the OrderedAggregate to keep track of which columns we have/have not seen.

### CODE REVIEW

File: pqarrow/arrowutils/groupranges.go [hunk 1]
```diff
@@ -149,6 +149,16 @@ func GetGroupsAndOrderedSetRanges(
 					return nil, nil, nil, err
 				}
 			}
+		case VirtualNullArray:
+			for j := 0; j < arr.Len(); j++ {
+				cmp, ok := nullComparison(curGroup[i] == nil, true)
+				if !ok {
+					panic("null comparison should always be valid")
```
Comment (metalmatze) [inline pqarrow/arrowutils/groupranges.go:156]: Cannot this be an error?

File: pqarrow/arrowutils/nullarray.go [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+func (n VirtualNullArray) IsNull(_ int) bool {
+	return true
+}
+
+func (n VirtualNullArray) IsValid(_ int) bool {
+	return false
+}
+
+func (n VirtualNullArray) Data() arrow.ArrayData {
+	panic("VirtualNullArray: Data not implemented")
```
Comment (brancz) [inline pqarrow/arrowutils/nullarray.go:65]: is this not needed by flight?

**Review by brancz — APPROVED** [review]
Nice work!

File: pqarrow/arrowutils/nullarray.go (hunk 2 above)
Comment (asubiotto) [reply pqarrow/arrowutils/nullarray.go:65]: This virtual null array is only used internally by operators (just the ordered agg) and materialized into a physical null array of a given type before being passed on downstream so would never be serialized by flight.

**Comment by asubiotto** [comment]
Looks like CI found a flake, will investigate and fix.

File: pqarrow/arrowutils/nullarray.go (hunk 2 above)
Comment (brancz) [reply pqarrow/arrowutils/nullarray.go:65]: gotcha, sounds good

**Comment by asubiotto** [comment]
The flake was due to expecting ordering due to map iteration. This is now fixed.

**Pull request merged**